src/utils/x509.rs

```diff
@@ -40,6 +40,14 @@ impl SecurityInfo {
             true
         }
     }
+
+    pub fn hostname(self) -> Option<String> {
```

Comment:
If this should really consume `self`, could use `self.dns_names.into_iter().next()` to avoid copy.

Reply:
Yes, it using only in "destructing context" -- we read info, then drop all but hostname, otherwise it require cloning on other side